### PR TITLE
Fix handling of non-trailing whitespace on last line

### LINKED ISSUE #1445: SA1028 falsely reports when [assembly: InternalsVisibleTo("...")] is used at the end of file
I have this file:

    using System.Reflection;
    using System.Runtime.CompilerServices;

    [assembly: AssemblyVersion("1.0.0.0")]
    [assembly: AssemblyFileVersion("1.0.0.0")]

    [assembly: InternalsVisibleTo("SomeLibrary")]

StyleCopAnalyzers falsely reports 

    Warning	SA1028	Code must not contain trailing whitespace

on line 7 (`[assembly: InternalsVisibleTo("Santase.Logic.Tests")]`) character 10 (the space between `[assembly:` and `InternalsVisibleTo("SomeLibrary")]`)

stylecopbug

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #1445

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1028UnitTests.cs [hunk 1]
```diff
@@ -216,6 +216,24 @@ public class SA1028UnitTests : CodeFixVerifier
...
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestWhitespaceBeforeClosingBraceAsync()
+        {
+            string testCode = new StringBuilder()
+                .AppendLine("class ClassName")
+                .AppendLine("{")
+                .Append(" }")
+                .ToString();
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1028UnitTests.cs]: :question: Why is the StringBuilder necessary?

**Comment by vweijsters** [comment]
:+1: Looks good besides a minor remark

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1028UnitTests.cs (hunk 1 above)
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1028UnitTests.cs]: Just following the pattern set for this file as part of #797.

Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1028UnitTests.cs]: I never looked at the complete file, just at your changes :blush:
I don't like the style, but consistency is the more important factor.

**Pull request merged**